README.md

```diff
@@ -1,2 +1,46 @@
-# persistent-volume-migrator
-Tool to migrate Kubernetes In-tree and Flex volumes to Ceph-CSI
+# What is persistent-volume-migrator?
+
+The persistent-volume-migrator tool aims to help migrate Flex RBD Volumes to [Ceph-CSI](https://github.com/ceph/ceph-csi) RBD Volumes.
+
+# Getting Started
+
+**NOTE**: The procedure will come with downtime while the applications are scaled down for the volume migration.
+## Prerequisite
+This guide assumes you have created a Ceph cluster with Flex and [Ceph-CSI](https://github.com/ceph/ceph-csi).
+1. We assume that you have healthy cluster i.e. `ceph status` shows `health: OK`
+2. Minimum ceph version should be Ceph-CSI supported
+3. Stop the application pods that are consuming the flex volume(s) that need to be converted.
+4. Ceph-CSI storageClass should be created in which you want to migrate.
+
+## Steps to Migrate
+1. Stop the application pods that are consuming the flex volume(s) that need to be converted.
+2. Create migrator pod
+   1. `kubectl create -f manifests/migrator.yaml`
+3. Download the binary from release
+4. Run the command to [migrate the PVC(s)](#migrate-the-pvcs)
+
+## Migrate the PVCs
+### Migrate a Single PVC
+
+```console
+pv-migrator --pvc=<pvc-name> --pvc-namespace=<pvc-namespace>
+   --destination-sc=<storageclass-name-to-migrate-in>
+   [--rook-namespace=rook-operator-namespace]
+   [--ceph-cluster-namespace=ceph-cluster-namespace]
+```
+   1. `--pvc`: **required**: reads name of the pvc to migrate
+   2. `--pvc-ns`: **required**: reads the namespace of the PVC which is going to migrate
+   3. `--destination-sc`: **required**: read the name of the storageclass in which you want mirgrate.
+   4. `--rook-ns`: **optional** namespace where the rook operator is running. **defalult: rook-ceph**.
+   5. `--ceph-cluster-ns`: **optional** namespace where the ceph cluster is running. **defalult: rook-ceph**.
```

Comment:
Maybe add a note just to highlight more that the user does not need to pass storageclass name when migrating a single PVC?

Reply:
added example command and output so I think it will be clear now.

Reply:
```suggestion
   5. `--ceph-cluster-ns`: **optional** namespace where the ceph cluster is running. ** default: rook-ceph**.
```